tests/Benchmarks/AzureBench.cs

```diff
@@ -88,5 +112,135 @@ static string CosmosDbBaselineFromMsdn(string key, string keyType, string verb,
 
         return result;
     }
+
+    public static bool TryWritePrimitives(Span<byte> output, Sha256 hash, string keyType, string verb, string resourceId, string resourceType, string tokenVersion, DateTime utc, out int bytesWritten)
+    {
+        int written, consumed, totalWritten = 0;
+        bytesWritten = 0;
+
+        Span<byte> buffer = stackalloc byte[AuthenticationHeaderBufferSize];
+
+        s_type.CopyTo(buffer);
+        totalWritten += s_type.Length;
+        var bufferSlice = buffer.Slice(totalWritten);
+
+        if (Encodings.Utf16.ToUtf8(keyType.AsReadOnlySpan().AsBytes(), bufferSlice, out consumed, out written) != OperationStatus.Done)
```

Comment:
We never use consumed anywhere in this method. Remove the local and replace with `out _`?

Reply:
There are multiple scoped in this code all using consumed. So each callsite would have to have it's own variable, which is probably inefficient.

Reply:
Regardless, you save a couple of instructions (an xor and mov) if you don't have the local and replace all function calls with `out _`:
![image](https://user-images.githubusercontent.com/6527137/32081166-9371977a-ba68-11e7-965d-a465405bdfa5.png)